vending-app/src/main/java/org/microg/vending/ui/WorkAppsActivity.kt

```diff
@@ -0,0 +1,390 @@
+import org.microg.vending.billing.core.GooglePlayApi.Companion.URL_FDFE
+import org.microg.vending.billing.core.GooglePlayApi.Companion.URL_ITEM_DETAILS
+import org.microg.vending.billing.core.GooglePlayApi.Companion.URL_PURCHASE
+import org.microg.vending.billing.core.HttpClient
+import org.microg.vending.billing.createDeviceEnvInfo
+import org.microg.vending.billing.proto.FailedResponse
+import org.microg.vending.billing.proto.GoogleApiResponse
+import org.microg.vending.enterprise.EnterpriseApp
+import org.microg.vending.delivery.requestDownloadUrls
+import org.microg.vending.enterprise.App
+import org.microg.vending.enterprise.AppState
+import org.microg.vending.enterprise.CommitingSession
+import org.microg.vending.enterprise.Downloading
+import org.microg.vending.enterprise.InstallError
+import org.microg.vending.enterprise.Installed
+import org.microg.vending.enterprise.NotCompatible
+import org.microg.vending.enterprise.NotInstalled
+import org.microg.vending.enterprise.Pending
+import org.microg.vending.enterprise.UpdateAvailable
+import org.microg.vending.enterprise.proto.AppInstallPolicy
+import org.microg.vending.proto.AppMeta
+import org.microg.vending.proto.GetItemsRequest
+import org.microg.vending.proto.RequestApp
+import org.microg.vending.proto.RequestItem
+import org.microg.vending.ui.components.EnterpriseList
+import org.microg.vending.ui.components.NetworkState
+import java.io.IOException
+import kotlin.random.Random
+
+@RequiresApi(android.os.Build.VERSION_CODES.LOLLIPOP)
+class WorkAppsActivity : ComponentActivity() {
+
+    private var apps: MutableMap<EnterpriseApp, AppState> = mutableStateMapOf()
+    private var networkState by mutableStateOf(NetworkState.ACTIVE)
+
+    private var auth: AuthData? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        enableEdgeToEdge()
+        super.onCreate(savedInstanceState)
+
+        ProfileManager.ensureInitialized(this)
+
+        val accountManager = AccountManager.get(this)
+        val accounts = accountManager.getAccountsByType(AuthConstants.WORK_ACCOUNT_TYPE)
+        if (accounts.isEmpty()) {
+            // Component should not be enabled; disable through receiver, and redirect to main activity
+            WorkAccountChangedReceiver().onReceive(this, null)
+            startActivity(Intent(this, MainActivity::class.java))
+            finish()
+        } else if (accounts.size > 1) {
+            Log.w(TAG, "Multiple work accounts found. This is unexpected and could point " +
+                    "towards misuse of the work account service API by the DPC.")
+        }
+        val account = accounts.first()
+
+        load(account)
+
+        setContent {
+            VendingUi(account,
+                install = { app, isUpdate -> Thread { runBlocking { install(app, isUpdate) } }.start() },
+                uninstall = { app -> Thread { runBlocking { uninstall(app) } }.start() }
```

Comment:
This should happen on a service to be independent of the lifecycle of this activity.